### PR TITLE
feat: Allow plugins & extra-files inputs to accept stringified JSON

### PR DESCRIPTION
**Pull request by phated**
This adds a JSON parsing step to handling `plugin` and `extra-files` input. This is useful because both of these often need to be configured (e.g. `{ "type": "yaml", "path": "path/to/file.yaml", "jsonpath": "$.json.path.to.field" }`).

Note: This still splits by multiline input **first** so the JSON has to be a single-line string. The best way to produce JSON is through the GHA [`toJSON` helper](https://docs.github.com/en/actions/learn-github-actions/expressions#tojson).

Not sure if this needs any tests because the `plugins` and `extra-files` don't currently have any tests. Happy to add them if needed and you point out how you'd want to do them. Cheers 🍻

### CODE REVIEW

**Comment by phated** [comment]
cc @chingor13 @bcoe :heart:

**Review by bcoe — APPROVED** [review]

**Comment by bcoe** [comment]
@phated you just need to run `npm run fix`, to correct linting.

**Comment by phated** [comment]
@bcoe Sorry about that! Should be good now.

**Comment by henry-brisson** [comment]
Will this really work? `getOptionalMultilineInput` returns an array of strings. Passing such array to `JSON.parse` will **always raise**, and cause the original array to be returned, which will contain the unparsed JSON string, which is probably not what we want...

**Comment by phated** [comment]
Might need to iterate the array and parse each value then 🤔

File: index.js [hunk 1]
```diff
@@ -24,6 +24,18 @@ function getGitHubInput () {
...
+  if (!value) {
+    return value
+  }
+
+  try {
+    return JSON.parse(value)
+  } catch (e) {
+    return value
+  }
+}
```
Comment (eungkyu) [inline index.js:37]: I have also looked into this issue, and fixing it this way should not cause problems.

```
  return value.map(line => {
    try {
      return JSON.parse(line)
    } catch (e) {
      return line
    }
  })
```

Comment (eungkyu) [reply index.js:37]: Actually, I have already fixed it this way and are currently using it without problems.

File: index.js [hunk 2]
```diff
@@ -24,6 +24,18 @@ function getGitHubInput () {
...
+function safeParse (value) {
+  if (!value) {
+    return value
+  }
+
+  try {
+    return JSON.parse(value)
+  } catch (e) {
+    return value
+  }
```
Comment (phated) [inline index.js:36]:
```suggestion
  return value.map(line => {
    try {
      return JSON.parse(line)
    } catch (e) {
      return line
    }
  })
```

File: index.js (hunk 1 above)
Comment (phated) [reply index.js]: Thanks! I've committed this through github UI

**Review by eungkyu — APPROVED** [review]

**Comment by JohnCampionJr** [comment]
Can we get this merged?

**Comment by gerynugrh** [comment]
> Can we get this merged? (I figured out my problems LOL)
> 
> In case this helps anybody else:
> 
> ```yaml
>           extra-files: |
>             { "type": "xml", "path": "src/Test.csproj", "xpath": "//project/propertygroup/version" }
>             { "type": "xml", "path": "src/Directory.Build.props", "xpath": "//project/propertygroup/version" }
> ```

Hi @JohnCampionJr I face the same problem as you did. I notice that the two codes you were sharing are identical. Can you explain more about the mistake on code?

**Comment by prokopsimek** [comment]
@bcoe Any updates on this PR? I would welcome its completion.

**Comment by chingor13** [comment]
For complex configuration, please switch to the [manifest configuration](https://github.com/googleapis/release-please/blob/main/docs/manifest-releaser.md) built into release-please.

We really don't want to be supporting custom input parsing from yaml for every advanced configuration option.

**Pull request closed without merging**